packages/engine/Source/Scene/MetadataPicking.js

```diff
@@ -262,11 +263,52 @@ MetadataPicking.convertToObjectType = function (type, value) {
   return value;
 };
 
+/**
+ * Converts the given type into an raw value or array representation.
+ *
+ * For `VECn/MATn` types, the given value is converted into an array.
+ * For other types, the value is returned directly
+ *
+ * @param {string} type The `ClassProperty` type
+ * @param {any} value The input value
+ * @returns {any} The array representation
```

Comment:
Since this "type" is used repeatedly, I would suggest using [`typedef`](https://jsdoc.app/tags-typedef) to define it fro both convenience, accuracy, and specificity.

For instance, I think that `MetadataPicking.convertToObjectType` has incomplete types for the `value` parameter. 

And `Scene.prototype.pickMetadata` could be updated to use this more specific type instead of `any`.

Something like:
```
/**
 * A raw value or array representation of a metadata value
 * @typedef {(number|string|boolean|number[]|undefined)} MetadataType.packedValue
 */
```

I maybe would suggest doing the same for the instance values, like:
```
/**
 * An instance of a metadata value
 * @typedef {(number|string|boolean|Cartesian2|Cartesian3|Cartesian4|Matrix2|Matrix3|Matrix4|undefined)} MetadataType.instanceValue
 */
```

Reply:
This is somehow related to https://github.com/CesiumGS/cesium/pull/12075#discussion_r1777461460 

The type information for metadata is difficult. You mentioned
`(number|string|boolean|Cartesian2|Cartesian3|Cartesian4|Matrix2|Matrix3|Matrix4|undefined)`

But in reality, in many places, it would be something like

`(bigint|number|string|boolean|Cartesian2|Cartesian3|Cartesian4|Matrix2|Matrix3|Matrix4|bigint[]|number[]|string[]|boolean[]|Cartesian2[]|Cartesian3[]|Cartesian4[]|Matrix2[]|Matrix3[]|Matrix4[]|undefined)`

(Give and take a few, depending on the subtle limitations here and there that are _highly_ specific for the context (e.g. for property textures, or implementations that are "incomplete" in terms of types that actually are supported)

It would be unfortunate if this tempted people to not even _trying_ to document things properly, but just use the
```
/**
 * @private
 */
```
that currently appears 349 times in the codebase.

I'd _like_ to be more specific. And I can _try_ to be more specific. But thoughness takes time...

Reply:
No need to fix this _everywhere_ in the code base for this PR. Would it be sensible to create the relevant type definitions with our best list of possible types and use them in the scope of this PR?

We can submit a cleanup issue that can cover the remaining cases.

I'm harping on this because the types are exposed through a public API, and the typings will help users of the library, especially those using typescript.

Reply:
Are you opposed to keeping the `{any}` in this (private!) part for now, and tracking possible refinements of the JSDoc for better TypeScript-consumabiltiy in a dedicated issue?